mlflow/projects/__init__.py

```diff
@@ -73,13 +74,18 @@ def _run(uri, entry_point="main", version=None, parameters=None, experiment_id=N
             remote_run=active_run,
             uri=uri, entry_point=entry_point, work_dir=work_dir, parameters=parameters,
             experiment_id=exp_id, cluster_spec=cluster_spec)
+
     elif mode == "local" or mode is None:
         # Synchronously create a conda environment (even though this may take some time) to avoid
         # failures due to multiple concurrent attempts to create the same conda env.
         conda_env_name = _get_or_create_conda_env(project.conda_env_path) if use_conda else None
         # In blocking mode, run the entry point command in blocking fashion, sending status updates
         # to the tracking server when finished. Note that the run state may not be persisted to the
         # tracking server if interrupted
+        git_remote_url = _get_remote_repo_url(work_dir)
+        if git_remote_url is not None:
+            tracking.MlflowClient().set_tag(active_run.info.run_uuid, MLFLOW_GIT_REPO_URL,
+                                            git_remote_url)
```

Comment:
Hmm, doesn't this means that the tags will only be recorded if mode = local? I think it should be possible to fix the test that was failing to handle the new tag too. The code there was just checking the order tags are added, so you can see when this one is added (maybe it's before or after some other tags) and do it then.

Reply:
It seems like the right place to put this is actually after `if _is_valid_branch_name(work_dir, version)`. Maybe It won't even clash with that other stuff then.